zone/bot.cpp

```diff
@@ -11443,7 +11446,7 @@ bool Bot::AttemptForcedCastSpell(Mob* tar, uint16 spell_id, bool is_disc) {
 		return false;
 	}
 
-	if (!DoLosChecks(tar)) {
+	if (!HasLoS() && !DoLosChecks(tar)) {
```

Comment:
Seems strange to not be able to rely on the member variable for HasLoS and also check / calculate anyways here. We should just simplify down to using DoLosChecks and do other things to measure how heavy LOS checks are with bot code paths and look for other means to optimize costs